### PR TITLE
feat: Email signature API support

### LINKED ISSUE #3799: feat: API for saving Email Signature with profile settings
**Context**
This is a follow up/breakdown of #3378. 
With every agent in chatwoot we need to support email signatures. The email signature would come under profile settings. This would need changes to  `PUT /api/v1/profile` API. With support for `email_signature` and `email_signature_enabled` keys.

`email_signature` - Would need to support HTML text also. Would also need to take care sanitation here

`email_signature_enabled` - If this is set to true, we would append the html value in `email_signature` to every outgoing email of the agent.

### PR DESCRIPTION
**Pull request by tejaswinichile**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #3799 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Added test cases to save and accept email signatures.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by tejaswinichile** [comment]
email_signature_api

@nithindavid Let's have a call when you are back, I have added basic API changes to save email_signature and email_signature_enabled.

**Comment by nithindavid** [comment]
@tejaswinichile I noticed the `email_signature_enabled` doesn't get toggled to `false` if the `email_signature` is not empty. Can we make it independent of each other? So that email signature will remain even if it is disabled ?

File: db/schema.rb [hunk 1]
```diff
@@ -747,6 +747,8 @@
     t.jsonb "ui_settings", default: {}
     t.jsonb "custom_attributes", default: {}
     t.string "type"
+    t.boolean "email_signature_enabled", default: false, null: false
```
Comment (pranavrajs) [inline db/schema.rb:750]: Can we use a generic name here? We can use the same signature across channels.

Comment (tejaswinichile) [reply db/schema.rb:750]: @pranavrajs Just `signature_enabled` will work?

**Comment by tejaswinichile** [comment]
> I noticed the email_signature_enabled doesn't get toggled to false if the email_signature is not empty. Can we make it independent of each other?

Oh okay, that's because of the validation rule, let me change that.

@nithindavid

File: db/schema.rb (hunk 1 above)
Comment (pranavrajs) [reply db/schema.rb:750]: let us put it as message_signature and message_signature_enabled

**Review by nithindavid — APPROVED** [review]

**Comment by tejaswinichile** [comment]
Merging this to save the email signature at the profile endpoint.

**Pull request merged**